### PR TITLE
oop_principles / it-academy.by / JC1-117-22 / Aliaksei Kharko / Done tasks and tests for Project: 'Cat's Home' and 'Battle Ground'

### PR DESCRIPTION
**Pull request by KharkoAO**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/Cat.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package com.rakovets.course.java.core.practice.oop_principles.cats_home;
+
+class Cat {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/Cat.java:3]:
```suggestion
public class Cat {
```

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/CatsTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return Stream.of(
+                Arguments.of("Mr. Paw", "Mr. Paw meow-meow purr"),
+                Arguments.of("Big Cat", "Big Cat meow-meow purr"),
+                Arguments.of("Buddy", "Buddy meow-meow purr")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideArgumentsForSetAndGetNameMewAndPurrMethods")
+    void testSetAndGetNameMewAndPurrMethods(String name, String expected) {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/CatsTest.java:24]: separate tests

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/PersianTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package com.rakovets.course.java.core.practice.oop_principles.cats_home;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class PersianTest {
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/PersianTest.java:10]: use one test's class for Cats

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Enemy.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        }
+        return getHealth();
+    }
+
+    public boolean isAlive() {
+        return getHealth() > 0;
+    }
+
+    public int attackHero(Hero hero) {
+        return attackHero(hero);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Enemy.java:22]: What is it?

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Archer.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    public Archer(String name, int health) {
+        super(name, health);
+    }
+
+    @Override
+    public int attackEnemy(Enemy enemy) {
+        final int CRITICAL_ARROW = 3;
+        int damage = 60;
+
+        System.out.println("Archer attacks!");
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Archer.java:13]: remove it from prod code

**Review by rakovets — CHANGES_REQUESTED** [review]
You need fix and rewrite all tests

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/Cat.java (hunk 1 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/Cat.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Archer.java (hunk 5 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Archer.java]: This fixed

File: src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Enemy.java (hunk 4 above)
Comment (KharkoAO) [reply src/main/java/com/rakovets/course/java/core/practice/oop_principles/battle_groud/Enemy.java]: This fixed

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/CatsTest.java (hunk 2 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/CatsTest.java]: This fixed

File: src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/PersianTest.java (hunk 3 above)
Comment (KharkoAO) [reply src/test/java/com/rakovets/course/java/core/practice/oop_principles/cats_home/PersianTest.java]: This fixed

**Comment by rakovets** [comment]
outdated

**Pull request closed without merging**